src/vendor/perfherder.js

```diff
@@ -126,11 +132,13 @@ const getFramework = async ({ repo, framework }) => {
             id: meta.id,
             framework: meta.framework_id,
             repo,
+            isPlayback,
           };
         })
         .toArray();
 
       PERFHERDER.signatures.push(...clean);
+      PLAYBACK.push(...(clean.filter(({ isPlayback }) => isPlayback)));
```

Comment:
This is very specific.  I see two logical paths weaved through the code; the original and the `isPlayback` paths.  The `isPlayback` subtitly depends on the original path; preventing us from improving the original; like making it faster in general. If we keep this code, then we must add tests for the `isPlayback` path so we do not beak it accidentally.  The original path did not need tests because it is used everywhere; its failure would be obvious on any page. If we move away from one logical path, we will need automated testing on those paths because a human will forget to test them all manually. 

Your specific solution does allude to a general solution: Filter the main list to make a short-list of tests.  Since the playback tests use a type of `suite` filtering (eg `suite: ["raptor-youtube-playback-fenix-live"]`), we can take advantage of that:

We can split a filter using the (badly named) `extract` function:

```
/*
 * Expecting props to be an array of N property names
 * Return a list of expression tuples; meant to represent disjunctive normal form
 * such that
 *   output[x][i] contains expression on props[i]
 *   output[x][N] is expressions on everything else
 */
const extract = (expression, props) => {
```

and use it like so

    suite_filter, residue_filter = extract(filter, ["suite"]);

we can make a general cache holding the complete list of signatures for every suite

    cache = {}

if the suite_filter looks like 

    suite_filter == {eq: {suite: suite_name}}
    suite_filter == {eq: {suite: ["raptor-youtube-playback-fenix-live"]}}

then we can jump to a the specific signature list:

    short_list = cache[suite_name]

if `cache[suite_name]` is missing, we can build it.  If there is no clean suite filter (rare, but it happens) then we can simply use the full `PERFHERDER.signatures`.

Your code already identifies the places that need to change; your code should be enhanced to handle `suite` in general.

Reply:
Thanks for the detailed review. I thought about a more general approach on this (in the end I forced this simple approach only for video playback tests), but your idea is a lot better than what I though about. I haven't had the time this week to work on it